src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -511,7 +510,7 @@ private Task ProcessCompilationUnitCompleted(CompilationUnitCompletedEvent compl
             }
         }
 
-        private async Task ExecuteCompilationActionsAsync(ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>> compilationActionsMap, CancellationToken cancellationToken)
+        private Task ExecuteCompilationActionsAsync(ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>> compilationActionsMap, CancellationToken cancellationToken)
```

Comment:
No, the Async suffix corresponds to the return type (and thus the manner in which the method is called), not to the manner in which it is implemented.